plugins/timertop/clienttimermodel.cpp

```diff
@@ -44,24 +48,60 @@ ClientTimerModel::~ClientTimerModel()
-        case TimerModel::MaxTimePerWakeupColumn:
-            return maxWakeupTimeToString(QSortFilterProxyModel::data(index, role).toUInt());
+    if (hasIndex(index.row(), index.column())) {
+        if (role == Qt::DisplayRole) {
+            switch (static_cast<TimerModel::Columns>(index.column())) {
+            case TimerModel::ColumnCount:
+                Q_ASSERT(false);
+            case TimerModel::ObjectNameColumn:
+            case TimerModel::TimerIdColumn:
+            case TimerModel::TotalWakeupsColumn:
+                // Use source model data
+                break;
+            case TimerModel::StateColumn:
+                return stateToString(QSortFilterProxyModel::data(index, role).toInt(),
+                                     QSortFilterProxyModel::data(index, TimerModel::TimerIntervalRole).toInt());
+            case TimerModel::WakeupsPerSecColumn:
+                return wakeupsPerSecToString(QSortFilterProxyModel::data(index, role).toReal());
+            case TimerModel::TimePerWakeupColumn:
+                return timePerWakeupToString(QSortFilterProxyModel::data(index, role).toReal());
+            case TimerModel::MaxTimePerWakeupColumn:
+                return maxWakeupTimeToString(QSortFilterProxyModel::data(index, role).toUInt());
+            }
+        } else if (role == Qt::ToolTipRole) {
+            const QModelIndex sibling = index.sibling(index.row(), TimerModel::ObjectNameColumn);
+            const TimerId::Type type = TimerId::Type(sibling.data(TimerModel::TimerTypeRole).toInt());
+            switch (type) {
+            case TimerId::InvalidType:
+                return tr("Invalid");
+            case TimerId::QQmlTimerType:
+                return tr("QQmlTimer");
+            case TimerId::QTimerType:
+                return tr("QTimer");
+            case TimerId::QObjectType:
+                return tr("Free Timer");
+            }
+        } else if (role == Qt::FontRole) {
+            const QModelIndex stateSibling = index.sibling(index.row(), TimerModel::StateColumn);
+            const TimerIdInfo::State state = TimerIdInfo::State(QSortFilterProxyModel::data(stateSibling, Qt::DisplayRole).toInt());
+            const QModelIndex typeSibling = index.sibling(index.row(), TimerModel::ObjectNameColumn);
+            const TimerId::Type type = TimerId::Type(typeSibling.data(TimerModel::TimerTypeRole).toInt());
+            QFont font = QApplication::font("QAbstractItemView");
+            font.setStrikeOut(type == TimerId::InvalidType || state == TimerIdInfo::InvalidState);
+            return QVariant::fromValue(font);
+        } else if(role == Qt::BackgroundRole) {
+            const QModelIndex sibling = index.sibling(index.row(), TimerModel::ObjectNameColumn);
+            const TimerId::Type type = TimerId::Type(sibling.data(TimerModel::TimerTypeRole).toInt());
+            switch (type) {
+            case TimerId::InvalidType:
+                return QColor(255, 0, 0, 80);
```

Comment:
Do these hardcoded colors work with light and dark color schemes?

Reply:
Tested on both, looks ok.